climpred/prediction.py

```diff
@@ -101,14 +162,37 @@ def compute_hindcast(
 
     Returns:
         skill (xarray object):
-            Predictability with main dimension ``lag``
+            Predictability with main dimension ``lag`` without dimension ``dim``
 
     """
+    if metric in PROBABILISTIC_METRICS:
+        if metric == 'e2r':
+            raise ValueError(
+                'Probabilistic metric `', metric, '` requires comparison `m2r`'
+            )
+        else:
+            stack = False
+    elif dim == 'init':
```

Comment:
Why is this an `elif`? Isn't this independent of the previous if statement?

Reply:
if probabilistic, then automatically dim='member' is used internally (doesnt need to be specified)
still, the user can also specify dim='member' for deterministic metrics to get a skill for each member and then do a mean over all members. the ordering of computation is different, but the results are fairly similar.
```python
compute_hindcast(hind, reconstruction, comparison='m2r',dim='member', metric=metric).mean('init').plot()
compute_hindcast(hind, reconstruction, comparison='e2r',dim='init', metric=metric).plot()
compute_hindcast(hind, reconstruction, comparison='m2r',dim='init', metric=metric).mean('member').plot()